lib/api/_loaders/element-global.js

```diff
@@ -89,7 +90,23 @@ class ElementGlobal {
 
     const byLocator = Locator.create(locator);
 
-    this.element = await this.transport.driver.wait(until.elementLocated(byLocator), this.timeout, null, this.retryInterval);
+    this.setPropertiesFromElement(locator);
+
+    const elements = await this.transport.driver.wait(until.elementsLocated(byLocator), this.timeout, null, this.retryInterval);
+
+    if (elements.length === 0) {
+      throw new Error(`Element ${byLocator} was not found.`);
+    }
+
+    if (typeof(this.index) != 'number'){
+      throw new Error(`Index ${this.index} is not of type number for locator: ${byLocator}`);
+    }
```

Comment:
We should move this condition here:
https://github.com/nightwatchjs/nightwatch/blob/4d799009a229cde905fee3ee8cf6f29b913dcd7c/lib/api/_loaders/element-global.js#L114-L116

And no need to throw an error if `index` is not of type number, we'll just silently ignore the `index`.